### PR TITLE
A1 benchmark example

### PR DESCRIPTION
**Pull request by danbarla**
This PR adds an A1 example in order to create a benchmark for future improvements on the factor graph.
The example includes a main.cpp which builds the factor graph, optimizes and writes the results to a .csv file.
The example inclues also a python simulation script which takes the values and simulates the robot motion.

### CODE REVIEW

**Comment by danbarla** [comment]
https://user-images.githubusercontent.com/67676161/143093504-359ff642-10b0-4ce0-8237-9d73267fd087.mp4

File: examples/example_a1_walking/main.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+using gtsam::Point3;
+using gtsam::Pose3;
+using gtsam::Rot3;
+using gtsam::Vector6;
+using gtsam::noiseModel::Isotropic;
+using gtsam::noiseModel::Unit;
+
+using namespace gtdynamics;
+
+// Returns a Trajectory object for a single spider walk cycle.
```
Comment (dellaert) [inline examples/example_a1_walking/main.cpp:40]: Spider? ;-)

**Review by dellaert — APPROVED** [review]
LGTM, but I thought we already had A1 files. Maybe in SDF models??

File: examples/example_a1_walking/main.cpp (hunk 1 above)
Comment (danbarla) [reply examples/example_a1_walking/main.cpp]: I had a little trouble with the a1 sdf which was in the repo already. It wouldn't load in pybullet at all.

**Review by dellaert — APPROVED** [review]
For future reference, you don't need to request a re-review if the changes were simple to implement and I already approved:-)

**Pull request merged**